### ISSUE TITLE
[Feature Request] Standby Killer for Line-out connected speakers

### ISSUE THREAD

**Issue by Dijker** [athombv/homey#1528]
**What is your Homey version (Settings → System)?**
Homey 1.x

**What did you try to achieve?**
Use speakers on Line-out, 
many Speakers switch off or to a standby when not used several minutes.
 
*Please include detailed steps how we can reproduce the issue*
Try to let Homey  say something after the speakers go to standby.

**What was the result?**
First seconds of the msg cant be heard as the speaker have to switch on first.

**What did you expect as result?**
To disable Power-saving anywhere, but as many speakers don't have the option you need to play something with an interval.
See the great workaround from @OpenMind_NL here: https://forum.athom.com/discussion/3153 

**Feature Request / suggestion**
Create an (experimental) checkbox in Settings - Speaker to disable Power-saving for external Line-Out connected speakers.
1) Once enabled Homey plays a 10 or 20 Hz tone for some short period whenever the speaker isn't used for  [Configurable] minutes.
2) Option 2: [Default On]: Only disable Power-saving  during Active Hours (Stops killing the Power-saver when nobody at Home or everyone at sleep to let the Speaker automatically switch to power-saving. Setting this to [Off] means speaker will be on 24/7.
3) Option 3: Predictive Power-up connected Speaker: Whenever the speaker may be in Power-saving (Speaker Not-used > [Configured Minutes]) and Ok-Homey is triggered play the tone to switch on the speaker even before Homey listened or has to respond. 
4) Option 4: Delayed speech: Whenever the speaker isn't used for more then the above  [Configurable] minutes first play the tone to switch on the speaker and delay the (spontaneous ) speech from Homey with [Configurable] seconds 

IMHO this option would fix the Start-up Problem introduced by the (EU mandatory?) Power-saving in many speakers.

**Comment by WeeJeWel** [comment]
You could create an app that does this by playing a sound file every x seconds.

**Comment by Dijker** [comment]
Hi @WeeJeWel 
Yeah I know it is possible with a flow, 
I guess for a good implementation it need to much info from and integration with the core and will take some time until anyone comes up with this App. 

Thanks for not instantly closing it, still hoping you come up one day with a msg like: "yesterday I sat on the toilet and this wasn't that much work..."  ;-)

**Comment by OpenMindNL** [comment]
@Dijker , I think Emile is right by not implementing it like this in Homey's software. The negative side of using this method would be that the speaker is consuming energy all day...

It would be better when the "wake-up-sound" is started a few seconds before Homey is going to speak. But this would involve adding delays to every flow where Homey speaks because it cannot be predicted in advance.
And it would only work when programmed in flows. For system reactions it's not possible to switch on the speakers in time.

@WeeJeWel , maybe it's possible to add a 'power-on' checkbox at speaker settings (or speech settings...) when line-out is used?
- When there has been no sound from line-out during the previous 9 minutes, Homey sends a very low frequency sine wave first. I think 3 seconds would be enough for most speakers.
- Homey's speech will be delayed once by 5-10 seconds (switch-on time of speaker. User setting?).

The next time Homey speaks within 9 minutes, this procedure is not necessary.

**Comment by OpenMindNL** [comment]
Hmm... Maybe my option is not that much different from Dijker's... It just sounds a lot easier to do :grin: ... and I think that the energy saving argument makes it a lot better than the "flow option".

**Comment by bvdbos** [comment]
This would seriously impact the UX for Homey... There's no "before Homey starts to speak" or "delay speech". Maybe when bug #1194 is fixed you could do this in a flow?

BTW, I think an mp3-file can be empty (no sound) too? https://github.com/anars/blank-audio

**Comment by Dijker** [comment]
@bvdbos Playing no sound doesn't keep the speaker on as Homey is already doing that a lot of time ;-) it just adds a pause between two other files played if they are played sequential!

@OpenMindNL  what part don't  you like or would you change  in my first suggestion after reading it completely ;-)

**Comment by OpenMindNL** [comment]
@Dijker , too many options for the regular user. It should be simple. Just switching the function on/off should be enough. Homey should figure out what has to be done next. 😉

**Comment by Dijker** [comment]
I Agree, with the defaults I suggested is is still an nice option. just enable/disable. I don't know if all speakers function with an 10 or 20 Hz tone, but we have to test!

**Comment by WeeJeWel** [comment]
If you (the community) tests many speakers with the same sample file, we actually might be willing to implement this :-)
